modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java

```diff
@@ -122,7 +122,7 @@ public AbstractTypeScriptClientCodegen() {
         typeMapping.put("integer", "number");
         typeMapping.put("Map", "any");
         typeMapping.put("map", "any");
-        typeMapping.put("date", "string");
+        typeMapping.put("date", "Date");
```

Comment:
This should not be changed for alle typescript generators, as Date serialization/deserialization logic may vary.
Moreover, the angular generator does not implement a deserialization logic, so it should remain string

Reply:
@macjohnny Ok, but why a string in that case hasnt ISO format?

Reply:
The consuming client / the backend needs to generate strings in iso format

Reply:
> The consuming client / the backend needs to generate strings in iso format

maybe then it is worth making a cast to ISO format? I don't mind having a type be a string if it is handled the same as the date format

Reply:
if you do not use ISO, how will you supply the timezone for the date?

Reply:
tell me, please, how can I build openapi-generator-cli for npm? Found and assembled only java version modules/openapi-generator-cli

Reply:
@macjohnny I want to build a fork version, tell me, please, how can I build openapi-generator-cli for npm? Found and assembled only java version. modules/openapi-generator-cli

Reply:
the string is not formatted by the client, it has to be passed by the consumer in the correct ISO format that the backend will understand.